doc/src/parts/user_guide.rst

```diff
@@ -3,472 +3,151 @@
-function definition. The placeholders are then automatically replaced by their corresponding
-*LaTeX* names. Due to the way *Python* implements string formatting, curly braces used in *LaTeX*
-need to be doubled, as shown in the code example.
-
-The full example additionally contains the creation of a contour plot. The corresponding lines are
-highlighted in the following example.
-
-.. bootstrap_collapsible::
-    :control_type: link
-    :control_text: python code
+Then there are multiple ways of displaying and using the fit results. The results can either be
+used directly inside a *Python*-script, can be printed to the terminal or can be
+:ref:`plotted <plotting>`.
+For further analysis, the :ref:`contours-profiler` is a very helpful tool to display parameter
+correlations.
 
-    .. literalinclude:: ../../../examples/002_model_functions/model_functions.py
-        :lines: 13-
-        :emphasize-lines: 43-45
+.. figure:: ../_static/img/kafe2_structure.png
+    :alt: General workflow with kafe2.
 
+.. _fitting:
 
-Example 3: Parameter Constraints
-================================
+Fitting
+=======
 
-The models used to describe physical phenomena usually depend on a multitude of parameters.
-However, for many experiments only one of the parameters is of actual interest to the experimenter.
-Still, because model parameters are generally not uncorrelated the experimenter has to factor in
-the nuisance parameters for their estimation of the parameter of interest.
+.. _minimizers:
 
-Historically this has been done by propagating the uncertainties of the nuisance parameters onto
-the y-axis of the data and then performing a fit with those uncertainties.
-Thanks to computers, however, this process can also be done numerically by applying parameter
-constraints. This example demonstrates the usage of those constraints in the kafe2 framework.
+Minimizers
+----------
+Currently the use of three different minimizers is supported. By default :py:mod:`iminuit` is
+used. If :py:mod:`iminuit` is not available, *kafe2* falls back to
+:py:obj:`scipy.optimize.minimize`.
 
-More specifically, this example will simulate the following experiment:
+The usage of a specific minimizer can be set during initialization of any
+:py:obj:`~.FitBase`-object with the `minimizer` keyword.
+Depending on the installed minimizers this can either be :code:`'iminuit'`, :code:`'scipy'` or
+:code:`'root'`.
 
-A steel ball of radius :math:`r` has been connected to the ceiling by a string of length :math:`l`,
-forming a pendulum. Due to earth's gravity providing a restoring force this system is a harmonic
-oscillator. Because of friction between the steel ball and the surrounding air the oscillator is
-also damped by the viscous damping coefficient :math:`c`.
+Additional keywords for the instantiation can be passed ass a :py:obj:`dict` via the
+`minimizer_kwargs` keyword when creating a :py:obj:`~.FitBase`-object.
```

Comment:
A user is not likely to understand what FitBase means.
How about this:
keyword when creating a fit object derived from :py:obj:`~.FitBase`.

Reply:
I was looking if it's possible to change the display of the link, so far I found no way of doing this. Using your solution for now.